bool value
int32 delta_x
int32 delta_y